src/plugins/intel_gpu/src/graph/kv_cache.cpp

```diff
@@ -31,6 +32,17 @@ std::vector<layout> kv_cache_inst::calc_output_layouts(kv_cache_node const& /*no
 
     std::vector<ShapeType> input_shapes = {impl_param.get_input_layout(0).get<ShapeType>(),
                                            impl_param.get_input_layout(1).get<ShapeType>()};
+                                     
+    std::unordered_map<size_t, ov::Tensor> const_data;
+    if (desc->update_kv) {
+        if(impl_param.memory_deps.count(2) > 0)
```

Comment:
is there a case where `desc->update_kv is true` and `deps.count(2) <= 0`? If not, what about replacing it with `OPENVINO_ASSERT`